lib/netext/httpext/transport.go

```diff
@@ -80,75 +80,51 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 		trail:             trail,
 	}
 
-	tags := t.tags
+	ctm := t.tagsAndMeta.Clone()
```

Comment:
I believe the context of the usage of this variable is big enough to give it a full name like `clonedTagsAndMetadata` (or just a `clonedTags`, or even better to call it `sampleTags` which I guess they are) since it can be essential to know downstream.

Reply:
IIRC, `ctm` is shorthand for `currentTagsAndMetadata`, so we can just use the `tagsAndMeta` here. 

Cloning is not the important bit here, this basically equivalent to `vu.State().Tags.GetCurrentValues()` (without the locking). We are just making a copy of the base `TagsAndMeta` object that is in `transport` because we might emit more than one sample (e.g. due to redirects). And the `transport` needs a custom copy of the tags+meta because its values can be modified from the script, e.g. `http.get(url, {tags: {foo: 'bar'}}).

Reply:
https://github.com/grafana/k6/pull/2726/commits/2428c114bbe91aff3e1685e6f47f1453aa561949